luskan/src/main/java/org/opencadc/luskan/LuskanConfig.java

```diff
@@ -0,0 +1,123 @@
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.util.PropertiesReader;
+
+public class LuskanConfig {
+
+    // config keys
+    private static final String LUSKAN_KEY = LuskanConfig.class.getPackage().getName();
+    static final String URI_KEY = LUSKAN_KEY + ".resourceID";
+    static final String TMPDIR_KEY = LUSKAN_KEY + ".resultsDir";
+    static final String STORAGE_SITE_KEY = LUSKAN_KEY + ".isStorageSite";
+
+    public LuskanConfig() { }
+
+    /**
+     * Read config file and verify that all required entries are present.
+     *
+     * @return MultiValuedProperties containing the application config
+     * @throws IllegalStateException if required config items are missing
+     */
+    static MultiValuedProperties getConfig() {
+        PropertiesReader r = new PropertiesReader("luskan.properties");
+        MultiValuedProperties props = r.getAllProperties();
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("incomplete config: ");
+        boolean ok = true;
+
+        String suri = props.getFirstPropertyValue(URI_KEY);
+        sb.append("\n\t").append(URI_KEY);
+        if (suri == null) {
+            sb.append("MISSING");
+            ok = false;
+        } else {
+            sb.append("OK");
+        }
+
+        String srd = props.getFirstPropertyValue(TMPDIR_KEY);
+        sb.append("\n\t").append(TMPDIR_KEY);
+        if (srd == null) {
+            sb.append("MISSING");
+            ok = false;
+        } else {
+            sb.append("OK");
+        }
+
```

Comment:
README implies that STORAGE_SITE_KEY is mandatory - check for it here and add to failure message. Also validate the value with Boolean.parseBoolean

Reply:
If it's mandatory that should be validated here. But I'm not sure these checks are doing what is intended. Looking at the MultiValueProperties class, **getFirstPropertyValue(key)** will return null if the property doesn't exist. These checks are verifying that the properties have a value, not that they are set.

Reply:
For properties that have a string value, like resourceID, a null value is an error. But for a boolean, a null value defaults to false.

Reply:
It is mandatory. This is verifying that the config file is complete and throws an exception with a useful message if not.